src/App/Platforms/Android/Services/AutofillHandler.cs

```diff
@@ -43,9 +43,28 @@ public bool CredentialProviderServiceEnabled()
             {
                 return false;
             }
+
             try
             {
-                // TODO - find a way to programmatically check if the credential provider service is enabled
+                var activity = (MainActivity)Platform.CurrentActivity;
+                if (activity == null)
+                {
+                    return false;
+                }
+
+                var credManager = activity.GetSystemService(Java.Lang.Class.FromType(typeof(CredentialManager))) as CredentialManager;
+                if (credManager == null)
+                {
+                    return false;
+                }
+
+                var credentialProviderServiceComponentName = new ComponentName(activity, Java.Lang.Class.FromType(typeof(CredentialProviderService)));
+                return credManager.IsEnabledCredentialProviderService(credentialProviderServiceComponentName);
+            }
+            catch (Java.Lang.NullPointerException) 
+            {
+                //TODO: CredentialManager API is not working fully and may return a NullPointerException even if the CredentialProviderService is working and enabled
```

Comment:
With the recent changes to the Settings structure, this is not a TODO anymore right?

Reply:
We can leave it, just remove the "TODO:". While commenting this, noticed I missed another one later in this file

`// TODO - find a way to programmatically disable the provider service, or take the user to the settings page where they can do it`

Reply:
All done